tls/s2n_quic_support.h

```diff
@@ -47,3 +47,30 @@ S2N_API int s2n_connection_set_quic_transport_parameters(struct s2n_connection *
  */
 S2N_API int s2n_connection_get_quic_transport_parameters(struct s2n_connection *conn,
         const uint8_t **data_buffer, uint16_t *data_len);
+
+typedef enum {
+    S2N_CLIENT_EARLY_TRAFFIC_SECRET = 0,
+    S2N_CLIENT_HANDSHAKE_TRAFFIC_SECRET,
+    S2N_SERVER_HANDSHAKE_TRAFFIC_SECRET,
+    S2N_CLIENT_APPLICATION_TRAFFIC_SECRET,
+    S2N_SERVER_APPLICATION_TRAFFIC_SECRET,
+} s2n_secret_type_t;
+
+/*
+ * Called when S2N begins using a new key.
+ *
+ * The memory pointed to by "secret" will be wiped after this method returns and should be copied by
+ * the application if necessary. The application should also be very careful managing the memory and
+ * lifespan of the secret: if the secret is compromised, TLS is compromised.
+ */
+typedef int (*s2n_secret_cb) (void* context, struct s2n_connection *conn,
```

Comment:
I'm assuming we'll need to call `s2n_connection_get_cipher` in order to know which cipher they should be used with, yes?

Reply:
Correct! I could make that an argument, but it seems more flexible to just let customers get the info they need from the connection APIs.